commons/src/main/java/io/aiven/kafka/connect/common/config/AivenCommonConfig.java

```diff
@@ -16,217 +16,35 @@
-    private static final String GROUP_FORMAT = "Format";
-    private static final String DEFAULT_FILENAME_TEMPLATE = "{{topic}}-{{partition}}-{{start_offset}}";
-
-    private static final String GROUP_RETRY_BACKOFF_POLICY = "Retry backoff policy";
-    public static final String KAFKA_RETRY_BACKOFF_MS_CONFIG = "kafka.retry.backoff.ms";
+/**
+ * @deprecated Use {@link SinkCommonConfig} instead
+ */
+@Deprecated
+public class AivenCommonConfig extends SinkCommonConfig {
 
     protected AivenCommonConfig(final ConfigDef definition, final Map<?, ?> originals) {
         super(definition, originals);
-        // TODO: calls getOutputFields, can be overridden in subclasses.
-        validate(); // NOPMD ConstructorCallsOverridableMethod
-    }
-
-    private void validate() {
-        // Special checks for output json envelope config.
-        final List<OutputField> outputFields = getOutputFields();
-        final Boolean outputEnvelopConfig = envelopeEnabled();
-        if (!outputEnvelopConfig && outputFields.toArray().length != 1) {
-            final String msg = String.format("When %s is %s, %s must contain only one field",
-                    FORMAT_OUTPUT_ENVELOPE_CONFIG, false, FORMAT_OUTPUT_FIELDS_CONFIG);
-            throw new ConfigException(msg);
-        }
-        validateKeyFilenameTemplate();
     }
 
     protected static void addKafkaBackoffPolicy(final ConfigDef configDef) {
-        configDef.define(KAFKA_RETRY_BACKOFF_MS_CONFIG, ConfigDef.Type.LONG, null, new ConfigDef.Validator() {
-
-            final long maximumBackoffPolicy = TimeUnit.HOURS.toMillis(24);
-
-            @Override
-            public void ensureValid(final String name, final Object value) {
-                if (Objects.isNull(value)) {
-                    return;
-                }
-                assert value instanceof Long;
-                final var longValue = (Long) value;
-                if (longValue < 0) {
-                    throw new ConfigException(name, value, "Value must be at least 0");
-                } else if (longValue > maximumBackoffPolicy) {
-                    throw new ConfigException(name, value,
-                            "Value must be no more than " + maximumBackoffPolicy + " (24 hours)");
-                }
-            }
-        }, ConfigDef.Importance.MEDIUM,
-                "The retry backoff in milliseconds. "
-                        + "This config is used to notify Kafka Connect to retry delivering a message batch or "
-                        + "performing recovery in case of transient exceptions. Maximum value is "
-                        + TimeUnit.HOURS.toMillis(24) + " (24 hours).",
-                GROUP_RETRY_BACKOFF_POLICY, 1, ConfigDef.Width.NONE, KAFKA_RETRY_BACKOFF_MS_CONFIG);
-    }
-
-    public Long getKafkaRetryBackoffMs() {
-        return getLong(KAFKA_RETRY_BACKOFF_MS_CONFIG);
+        SinkCommonConfig.addKafkaBackoffPolicy(configDef);
```

Comment:
Is it possible to keep AivenCommonConfig independent of sink and source configs ?

Reply:
No that would lead to a large amount of duplication between AivenCommonConfig and the SinkCommonConfig and CommonConfig classes and multiple locations to maintain this code which we are trying to avoid.